devtools/x-lint/src/runner.rs

```diff
@@ -239,7 +239,8 @@ impl<'cfg> LintEngine<'cfg> {
     // ---
 
     fn file_list(&self) -> Result<impl Iterator<Item = &'cfg Utf8Path> + 'cfg> {
-        let tracked_files = self.config.core.git_cli().tracked_files()?;
+        let git_cli = self.config.core.git_cli()?;
```

Comment:
Could you add a `.with_context()` to this?

Reply:
So this are bubbling up system errors not anyhow errors, from the core git_cli new method, your asking me to switch all the calls to anyhow, correct?

Reply:
Yeah :)

Reply:
Or at least switch them at some boundary. I guess this one slightly less so, hmm.

Reply:
Looks like to make this change I'll need to allow an anyhow error to be treated as a x_core::SystemError:

```
error[E0277]: `?` couldn't convert the error to `x_core::SystemError`
   --> devtools/x-lint/src/runner.rs:245:11
    |
245 |         })?;
    |           ^ the trait `From<anyhow::Error>` is not implemented for `x_core::SystemError`
    |
    = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
    = note: required by `from`
```

Not opposed, but the scope keeps growing.

Reply:
Can't find a way to make this different Result hierarchies play nice, and not about to change the error handling everywhere in the linting logic.    If you have a test you'd like me to run to prove xclippy's still good let me know.

Reply:
ahh never mind then